### PR TITLE
Texture analysis using Haralick Descriptors for Computer Vision tasks

### PR DESCRIPTION
**Pull request by rzimmerdev**
### Describe your change:
I've added a series of methods for calculating the Haralick descriptors, as well as some encapsulating methods for comparing textures between a baseline image and a sequence of comparison images.

* [x] Add an algorithm?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by rzimmerdev** [comment]
Still in progress

File: computer_vision/haralick_descriptors.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
+import imageio.v2 as imageio
+import numpy as np
+
+
+def rmse(original: np.ndarray, reference: np.ndarray) -> float:
```
Comment (cclauss) [inline computer_vision/haralick_descriptors.py:5]:
```suggestion
def root_mean_square_error(original: np.ndarray, reference: np.ndarray) -> float:
```

File: computer_vision/haralick_descriptors.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
+import imageio.v2 as imageio
```
Comment (cclauss) [inline computer_vision/haralick_descriptors.py:1]:
```suggestion
"""
https://en.wikipedia.org/wiki/Image_texture
https://en.wikipedia.org/wiki/Co-occurrence_matrix#Application_to_image_analysis
"""

import imageio.v2 as imageio
```

File: computer_vision/haralick_descriptors.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+
+def binarize(image: np.ndarray, threshold_value) -> np.ndarray:
+    """
+    Binarizes a grayscale image based on a given threshold value,
+    setting values to 1 or 0 accordingly.
+    """
+    binarized = np.where(image > threshold_value, 1, 0)
+
+    return binarized
```
Comment (cclauss) [inline computer_vision/haralick_descriptors.py:65]:
```suggestion
    return np.where(image > threshold_value, 1, 0)
```

File: computer_vision/haralick_descriptors.py [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+    return transformed
+
+
+def opening_filter(image: np.ndarray, kernel: np.ndarray = None) -> np.ndarray:
+    """
+    Opening filter, defined as the sequence of
+    erosion and then a dilation filter on the same image.
+    """
+    if kernel is None:
```
Comment (cclauss) [inline computer_vision/haralick_descriptors.py:116]: In this instance, all falsey values should be treated the same way as None.
```suggestion
    if not kernel:
```

File: computer_vision/haralick_descriptors.py [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+
+    return transform(transform(image, "dilation", kernel), "erosion", kernel)
+
+
+def closing_filter(image: np.ndarray, kernel: np.ndarray = None) -> np.ndarray:
+    """
+    Opening filter, defined as the sequence of
+    dilation and then erosion filter on the same image.
+    """
+    if kernel is None:
```
Comment (cclauss) [inline computer_vision/haralick_descriptors.py:127]:
```suggestion
    if not kernel:
```

File: computer_vision/haralick_descriptors.py [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+        maximum_prob,
+        correlation.sum(),
+        energy.sum(),
+        contrast.sum(),
+        dissimilarity.sum(),
+        inverse_difference.sum(),
+        homogeneity.sum(),
+        entropy.sum(),
+    ]
+    return descriptors
```
Comment (cclauss) [inline computer_vision/haralick_descriptors.py:213]:
```suggestion
    return [
        maximum_prob,
        correlation.sum(),
        energy.sum(),
        contrast.sum(),
        dissimilarity.sum(),
        inverse_difference.sum(),
        homogeneity.sum(),
        entropy.sum(),
    ]
```

File: computer_vision/haralick_descriptors.py [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+    array([[ 0,         0.111111,  0.222222],
+           [ 0.333333,  0.444444,  1.      ]], dtype=float32)
+    >>> normalize_image(np.array([[4, 4, 3], [1, 7, 2]]))
+
+    """
+    normalized = (image - np.min(image)) / (np.max(image) - np.min(image)) * cap
+    return normalized.astype(np.uint8)
+
+
+def normalize_array(array: np.ndarray, cap: float = 1) -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (cclauss) [reply computer_vision/haralick_descriptors.py:39]: Please examine the functions than have not doctests to see if we can define a simple 2x2 or 3x3 `test_matrix` that can be passed to them to add a basic doctest.

**Comment by rzimmerdev** [comment]
Hi there, I finished the suggested modifications and polished the docstrings.

**Pull request closed without merging**